src/redist/targets/GenerateBundledVersions.targets

```diff
@@ -1,7 +1,36 @@
 <Project>
 
+  <!-- Used for creating productVersion and productCommit text files. -->
+  <UsingTask
+    TaskName="WriteTextToFile"
+    TaskFactory="RoslynCodeTaskFactory"
+    AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
+    <ParameterGroup>
+      <Path Required="true" />
+      <Content Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Code Type="Fragment" Language="cs">
+        var dir = System.IO.Path.GetDirectoryName(Path);
+        if(!string.IsNullOrEmpty(dir))
+          System.IO.Directory.CreateDirectory(dir);
+        System.IO.File.WriteAllText(Path, Content);
+      </Code>
+    </Task>
+  </UsingTask>
+
   <Target Name="GenerateBundledVersions"
-          DependsOnTargets="GenerateBundledVersionsProps;GenerateBundledCliToolsProps" />
+          DependsOnTargets="GenerateBundledVersionsProps;GenerateBundledCliToolsProps" >
+          
+    <WriteTextToFile
```

Comment:
There's a built in task called WriteLinesToFile that does this already. Can you use that one?

Reply:
I did use it in earlier commits, but it adds a new line to the end that cannot be prevented. Since you mentioned you wanted to pipe the contents directly into install script, I used this approach. 

Should I change it back? Would you be OK with a new line character at the end?

Reply:
Yeah newline should be fine.